go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -569,13 +569,37 @@ func (npw *nodePortWatcher) UpdateService(old, new *kapi.Service) {
 	}
 }
 
+// deleteConntrackForClusterIP deletes the conntrack entries corresponding to the clusterIP of
+// the provided service
+func deleteConntrackForClusterIP(service *kapi.Service) error {
+	clusterIPs := service.Spec.ClusterIPs
+	if len(clusterIPs) == 0 {
+		clusterIPs = append(clusterIPs, service.Spec.ClusterIP) // handle old clusters w/o v6 support
+	}
+	for _, clusterIP := range clusterIPs {
+		for _, svcPort := range service.Spec.Ports {
+			err := deleteConntrack(clusterIP, svcPort.Port, svcPort.Protocol)
```

Comment:
I noticed in this function path that it will eventually end up creating an ip filter: netlink.ConntrackReplyAnyIP, which will:
// Match source or destination reply IP

Is that the desire here? I thought cluster ip would be the original dest IP? Like:
	ConntrackOrigDstIP                         // -orig-dst ip    Destination address from original direction

Maybe you can post the original 2 or 3 conntrack entries from the bz so we can compare?

Reply:
hmm yeah good question; I think we want the match on the reply tuple.

here is an example for a single connection state:

Traffic Scenario: Host  -> Load Balancer service -> backend pod

LB VIP: 172.31.249.39
clusterIP: 172.30.151.15 
backend pod: 10.131.0.27
srcIP: 172.31.249.96

this PR will delete these entries, essentially what we want to do is the equivalent of `conntrack -D -r <clusterIP>`
```
[DESTROY] sctp     132 src=172.31.249.96 dst=172.30.151.15 sport=47541 dport=30102 src=172.30.151.15 dst=169.254.169.2 sport=30102 dport=47541 [ASSURED] zone=64001
[DESTROY] sctp     132 src=172.31.249.96 dst=172.31.249.39 sport=47541 dport=30102 src=172.30.151.15 dst=172.31.249.96 sport=30102 dport=47541 [ASSURED]
[DESTROY] sctp     132 src=169.254.169.2 dst=172.30.151.15 sport=47541 dport=30102 src=10.131.0.27 dst=169.254.169.2 sport=30102 dport=47541 [ASSURED] zone=12
[DESTROY] sctp     132 src=169.254.169.2 dst=10.131.0.27 sport=47541 dport=30102 src=10.131.0.27 dst=100.64.0.5 sport=30102 dport=47541 [ASSURED]
[DESTROY] sctp     132 src=100.64.0.5 dst=10.131.0.27 sport=47541 dport=30102 src=10.131.0.27 dst=100.64.0.5 sport=30102 dport=47541 [ASSURED] zone=27
```

Reply:
`[DESTROY] sctp     132 src=172.31.249.96 dst=172.31.249.39 sport=47541 dport=30102 src=172.30.151.15 dst=172.31.249.96 sport=30102 dport=47541 [ASSURED]`
that's the key entry we need to remove. The DNAT from LB to clusterIP

Reply:
> I noticed in this function path that it will eventually end up creating an ip filter: netlink.ConntrackReplyAnyIP, which will: // Match source or destination reply IP
> 
> Is that the desire here? I thought cluster ip would be the original dest IP? Like: ConntrackOrigDstIP // -orig-dst ip Destination address from original direction
> 
yeah I do see your point you are thinking podORhost -> clusterIP traffic scenario, where the clusterIP will be in the dstIP of original tuple; but I don't think we need to care about that since if clusterIP goes away no one will be able to maintain an active connection to that.

Reply:
thanks for clarifying. We already attempt to flush the the endpoints when they are removed I think.